test/test_client.py

```diff
@@ -24,10 +24,14 @@
+            .format(version=kwargs.get('version', '1.24wmf17'),
+                    rights=json.dumps(kwargs.get('rights', [
+                        "createaccount", "read", "edit", "createpage", "createtalk",
+                        "editmyusercss", "editmyuserjs", "viewmywatchlist",
+                        "editmywatchlist", "viewmyprivateinfo", "editmyprivateinfo",
+                        "editmyoptions", "centralauth-merge", "abusefilter-view",
+                        "abusefilter-log", "translate", "vipsscaler-test", "upload"])))
 
```

Comment:
AGH GOD MY EYES

is this...actually better? can we manually redo it better...somehow?

Reply:
How about something like this instead? Still not necessarily _nice_, but much more readable. I've also dropped a bunch of non-default namespaces to make it more manageable.

```python
def metaResponse(self, version='1.24wmf17', rights=None):
    if rights is None:
        rights = [
            "createaccount", "read", "edit", "createpage", "createtalk",
            "editmyusercss", "editmyuserjs", "viewmywatchlist",
            "editmywatchlist", "viewmyprivateinfo", "editmyprivateinfo",
            "editmyoptions", "centralauth-merge", "abusefilter-view",
            "abusefilter-log", "translate", "vipsscaler-test", "upload"
        ]

    # @formatter:off
    namespaces = {
        -2: {"id": -2, "*": "Media", "canonical": "Media", "case": "first-letter"},
        -1: {"id": -1, "*": "Special", "canonical": "Special", "case": "first-letter"},
        0: {"id": 0, "*": "", "case": "first-letter", "content": ""},
        1: {"id": 1, "*": "Talk", "canonical": "Talk", "case": "first-letter", "subpages": ""},
        2: {"id": 2, "*": "User", "canonical": "User", "case": "first-letter", "subpages": ""},
        3: {"id": 3, "*": "User talk", "canonical": "User talk", "case": "first-letter", "subpages": ""},
        4: {"id": 4, "*": "Wikipedia", "canonical": "Project", "case": "first-letter", "subpages": ""},
        5: {"id": 5, "*": "Wikipedia talk", "canonical": "Project talk", "case": "first-letter", "subpages": ""},
        6: {"id": 6, "*": "File", "canonical": "File", "case": "first-letter"},
        7: {"id": 7, "*": "File talk", "canonical": "File talk", "case": "first-letter", "subpages": ""},
        8: {"id": 8, "*": "MediaWiki", "canonical": "MediaWiki", "case": "first-letter", "subpages": ""},
        9: {"id": 9, "*": "MediaWiki talk", "canonical": "MediaWiki talk", "case": "first-letter", "subpages": ""},
        10: {"id": 10, "*": "Template", "canonical": "Template", "case": "first-letter", "subpages": ""},
        11: {"id": 11, "*": "Template talk", "canonical": "Template talk", "case": "first-letter", "subpages": ""},
        12: {"id": 12, "*": "Help", "canonical": "Help", "case": "first-letter", "subpages": ""},
        13: {"id": 13, "*": "Help talk", "canonical": "Help talk", "case": "first-letter", "subpages": ""},
        14: {"id": 14, "*": "Category", "canonical": "Category", "case": "first-letter"},
        15: {"id": 15, "*": "Category talk", "canonical": "Category talk", "case": "first-letter", "subpages": ""},
    }
    # @formatter:on

    return {
        "query": {
            "general": {
                "generator": f"MediaWiki {version}"
            },
            "namespaces": namespaces,
            "userinfo": {
                "anon": "",
                "groups": ["*"],
                "id": 0,
                "name": "127.0.0.1",
                "rights": rights
            }
        }
    }
```

Reply:
yeah, I like that a lot more. thanks.